app/main.py

```diff
@@ -1,2 +1,112 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.hash_list = [None] * 8
+
+    def _set_item(self, key: Hashable, value: Any) -> None:
+        """This method is used inside the class for setting item"""
+
+        _hash = hash(key)
+        index = _hash % len(self.hash_list)
+
+        for _ in range(len(self.hash_list)):
+            if self.hash_list[index] is None \
+                    or self.hash_list[index][0] == key:
+                self.hash_list[index] = (key, _hash, value)
+                break
+            if index == len(self.hash_list) - 1:
+                index = -1
+            index += 1
```

Comment:
it would be best if you used a while loop to get optimal complexity

Reply:
done